muikku/src/main/java/fi/otavanopisto/muikku/session/SessionBackingBean.java

```diff
@@ -84,6 +107,99 @@ else if (user.getNickName() != null) {
 
       this.loggedUserId = sessionController.getLoggedUserEntity().getId();
       this.loggedUser = sessionController.getLoggedUser().toId();
+      
+      if (!sessionController.hasEnvironmentPermission(ForumResourcePermissionCollection.FORUM_ACCESSENVIRONMENTFORUM) || !sessionController.isActiveUser()) {
+        this.areaPermissions = null;
+      } else {
+        List<EnvironmentForumArea> forumAreas = forumController.listEnvironmentForums();
+        Map<Long, AreaPermission> areaPermissions = new HashMap<>();
+        
+        for (EnvironmentForumArea forumArea : forumAreas) {
+          AreaPermission areaPermission = new AreaPermission(
+              sessionController.hasPermission(ForumResourcePermissionCollection.FORUM_EDIT_ENVIRONMENT_MESSAGES, forumArea),
+              sessionController.hasPermission(ForumResourcePermissionCollection.FORUM_DELETE_ENVIRONMENT_MESSAGES, forumArea));
+          areaPermissions.put(forumArea.getId(), areaPermission );
+        }
+        
+        try {
+          this.areaPermissions = new ObjectMapper().writeValueAsString(areaPermissions);
+        } catch (JsonProcessingException e) {
+          this.areaPermissions = null;
+        }
+      }
+    } else {
+      this.areaPermissions = null;
+    }
+    
+    if (sessionController.isLoggedIn()) {
+    	UserEntity userEntity = sessionController.getLoggedUserEntity();
+        User user = userController.findUserByDataSourceAndIdentifier(sessionController.getLoggedUserSchoolDataSource(), sessionController.getLoggedUserIdentifier());
+        List<UserAddress> userAddresses = userController.listUserAddresses(user);
+        List<UserPhoneNumber> userPhoneNumbers = userController.listUserPhoneNumbers(user);
+        
+        displayName = user.getNickName() == null ? user.getDisplayName() : String.format("%s %s (%s)", user.getNickName(), user.getLastName(), user.getStudyProgrammeName());
+        
+        studyStartDate = user.getStudyStartDate();
```

Comment:
I'm not sure if these fields are used in more generic places than f.ex in guider. If they aren't, they probably should be in some other backing bean (having them in too generic place like this can cause overhead).

Reply:
Ok. We can look into it later.